Realm/RLMQueryUtil.hpp

```diff
@@ -46,8 +47,7 @@ realm::SortOrder RLMSortOrderFromDescriptors(RLMObjectSchema *objectSchema, NSAr
 // This macro validates predicate format with optional arguments
 #define RLM_VARARG(PREDICATE_FORMAT, ARGS) \
 va_start(ARGS, PREDICATE_FORMAT);          \
-va_end(ARGS);                              \
-if (PREDICATE_FORMAT && ![PREDICATE_FORMAT isKindOfClass:[NSString class]]) {         \
-    NSString *reason = @"predicate must be an NSString with optional format va_list"; \
-    [NSException exceptionWithName:RLMExceptionName reason:reason userInfo:nil];       \
-}
+va_end(ARGS);
```

Comment:
It looks like the behavior of this macro is wrong (and was wrong prior to this change). The expected usage of `va_list` to forward variadic arguments is:

```c++
va_list args;
va_start(predicate, args);
target_function(predicate, args);
va_end(args);
```

`va_end` is documented as invalidating the `va_list`, so our practice of `va_start` / `va_end`/ `target_function` is relying on undefined behavior.